perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -938,4 +938,18 @@ public static String cipherInput(String plainText, boolean decrypt) throws Inter
 
 	}
 
+	public static void checkDestinationType(Destination destination) throws WrongPatternException {
```

Comment:
Add there also possibility, that destination or type is null. Better than let throw NullPointerException.
You can throw InternalErrorException in this case.